### PR TITLE
Add automatic PR labels

### PR DESCRIPTION
**Pull request by bennettgoble**
Automatically label PRs with a handful of code-area based groups.

### CODE REVIEW

File: .github/labeler.yaml
Comment (nat-goodspeed) [inline .github/labeler.yaml:59]: Woof, my impulse here is to commit a script that will generate (most of) the `labeler.yaml` file, possibly appending the lines below from, say, a `labeler.yaml.in` file. This is a file I will _certainly_ forget to modify if in future I add a new `indra` subdirectory.

That said, we don't need to block this PR waiting for that.

**Review by nat-goodspeed — APPROVED** [review]

**Comment by bennettgoble** [comment]
> There is directories missing. At least
> maybe some more. Is this intended?

Yes, this is a fast first pass. More can be added as appropriate as a follow-up.

**Comment by bennettgoble** [comment]
> Please explain in a bit more detail what this accomplishes?

This change will result in PRs receiving a set of automatic labels based on the files that have been changed. This is useful for sifting through PRs after dozens/hundreds/thousands have been merged.

**Pull request merged**
